src/core/dfn.js

```diff
@@ -26,9 +34,49 @@ export function run() {
     }
 
     // Only add `lt`s that are different from the text content
-    if (titles.length === 1 && titles[0] === norm(dfn.textContent)) {
+    if (titles.length === 1 && linkingText === norm(dfn.textContent)) {
       return;
     }
     dfn.dataset.lt = titles.join("|");
   });
 }
+/**
+ *
+ * @param {string} title
+ * @param {HTMLElement} dfn
+ */
+function processAsInternalSlot(title, dfn) {
+  if (!dfn.dataset.hasOwnProperty("idl")) {
+    dfn.dataset.idl = "";
+  }
+
+  // Automatically use the closest data-dfn-for as the parent.
+  /** @type HTMLElement */
+  const parent = dfn.closest("[data-dfn-for]");
+  if (parent && dfn !== parent && parent.dataset.dfnFor) {
```

Comment:
I don't think it would be expensive. But I don't find it as clear. Maybe we could do a compromise... how about?:

```JS
if (parent?.dataset.dfnFor && dfn !== parent) {
```

Reply:
👍🏼  Though, `dfn !== parent` test before `dfnFor` one.